### ISSUE TITLE
flake8-datetimez (DTZ) should check datetime.combine

### ISSUE THREAD

**Issue by hofrob** [charliermarsh/ruff#2089]
One thing upfront, this is already missing in the `flake8` plugin. The [`datetime.combine`](https://docs.python.org/3.11/library/datetime.html?highlight=combine#datetime.datetime.combine) method accepts a timezone as an optional third parameter. It would make sense to create a rule that checks if that parameter is used, just like the other `flake8-datetimez` rules.

I'm aware that ruff is trying to copy the behavior of those plugins first. And there might be more methods that have an optional timezone parameter.

The point of this issue is to:

* decide if you want to expand the DTZ rules
* and if yes if you first want someone to do more research to collect all `datetime` methods that should be checked
* or if you preferred me first creating an issue/PR for the `flake8` plugin

# the actual rule

```py
import datetime
import zoneinfo

# Wrong:
offset_naive = datetime.datetime.combine(
    datetime.date(2023, 1, 22), datetime.datetime.min.time()
)
# this will create an offset-naive datetime

# Correct:
offset_aware = datetime.datetime.combine(
    datetime.date(2023, 1, 22),
    datetime.datetime.min.time(),
    zoneinfo.ZoneInfo("Europe/Vienna"),
)
# this will create an offset-aware datetime
```

**Comment by charliermarsh** [comment]
My general philosophy is that I'm happy to deviate from plugins in cases that are clearly wrong (false positives or false negatives), and/or in cases where the original behavior of the plugin can be restored via changes in settings. (For example, we changed Pylint's "magic constant" rule to ignore strings by default, but users can change that as a setting.)

In this case, it feels like a reasonable spot to add a new rule, regardless of the upstream behavior, since this is a false-negative. The main question for me is whether it should be in `DTZ`, or `RUF`. It's obviously more at-home in `DTZ`, but if we pick some rule code (like `DTZ008`), and then the plugin expands to include some other rule under that prefix, we'd be in a weird spot. (I guess this isn't that likely, given that their last release was in 2020.)

Perhaps we can find some pragmatic solution, like `DTZ901`? (Some plugins use the `9XX` range to indicate experimental or opinionated rules.)

**Comment by bluetech** [comment]
The suggested rule is not always applicable without type information, which ruff doesn't have. If the time (2nd) argument to `datetime.combine` is tz-aware, then the result is tz-aware, and the lint shouldn't fire.

**Comment by hofrob** [comment]
Ok, thanks for this! I wasn't aware(ha!) that you can have times with offset.

As much as I'd still want explicit timezone information in my code base, I know this might be too opinionated for ruff. So I'm fine with closing this issue if there are no other ideas!

**Comment by charliermarsh** [comment]
Ah ok. Yeah, that's gonna be hard to enforce reliably right now. Gonna close out for the moment.
